java/compaction-job-creation/src/main/java/sleeper/compaction/job/CompactionJobSerDe.java

```diff
@@ -32,15 +32,22 @@
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.UUID;
 
 /**
  * Serialised a {@link CompactionJob} to and from a JSON {@link String}.
  */
 public class CompactionJobSerDe {
     private final TablePropertiesProvider tablePropertiesProvider;
+    private final String taskId;
```

Comment:
CompactionJobSerDe doesn't need to know about the task ID. It's only used for reading from and writing to the queue. The compaction job only gets a task ID when it comes off the queue.